scripts/generate.sh

```diff
@@ -14,11 +14,19 @@ done
 
 # jar was downloaded from here https://repo1.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.34/
 
-if [[ $VERSION == *"develop"* ]]; then
-  SANITIZED_VERSION=$(sed 's/-develop//g' <<< $VERSION)
-  SANITIZED_VERSION=$(sed 's/\.[0-9]*$//g' <<< $SANITIZED_VERSION)
-  BUILD_VERSION=$(grep -o '[0-9]*$' <<< $VERSION)
-  VERSION="dev-$SANITIZED_VERSION-$BUILD_VERSION"
+VERSION=${VERSION//develop/dev}
+
+if [[ $VERSION =~ ^dev[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)$ ]]; then
+    VERSION="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.${BASH_REMATCH[3]}-dev.${BASH_REMATCH[4]}"
+elif [[ $VERSION =~ ^([0-9]+)[.-]([0-9]+)[.-]([0-9]+)[.-]dev[.-]([0-9]+)$ ]]; then
+    VERSION="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.${BASH_REMATCH[3]}-dev.${BASH_REMATCH[4]}"
+elif [[ $VERSION == dev-* ]]; then
+    _temp_part=${VERSION#dev-}
+    VERSION="${_temp_part//-/.}-dev"
+elif [[ $VERSION == *-dev* ]]; then
+    _temp_part=${VERSION#*-dev}
+    _temp_part=${_temp_part//-/.}
```

Comment:
For [stability](https://getcomposer.org/doc/articles/versions.md#stabilities) to be recognized as `dev`, the version needs to follow the [SemVer spec](https://semver.org/#spec-item-9) like `X.Y.Z-dev+M`. You can use [this regex](https://regex101.com/r/Ly7O1x/3/) and the snippet below to test this.

```php
<?php

use Composer\Semver\VersionParser;

include "../vendor/autoload.php";

var_dump(VersionParser::parseStability("6.5.0-dev.0"));
```